### ISSUE TITLE
Make it easier to extract values from the view elements for tests

### ISSUE THREAD

**Issue by jimbobbennett** [fsprojects/Fabulous#204]
The `ViewElement` created but the view function can be unit tested, but it's not always easy to do so. It would db nice to have some helper functions to extract aitributes/children etc.

**Comment by garethhubball** [comment]
I've been having a play around with this on my streams and come up with a hacky solution where the ViewElements have create called on them first and therefore become their Xamarin Forms equivalent:
https://github.com/garethhubball/TwitchDeck

There's also a helper method or two in there that will help me traverse the DOM. As the project grows I'm intending to do more for it and break out the test helper functions.

Given that there seems to be a mapping from string -> Xamarin Forms property would it be worth doing that kind of mapping in reverse in order to create a strongly-typed object to assert against?

**Comment by TimLariviere** [comment]
I'm currently looking for a way to generate those helpers as part of the Generator output, along #159.

I'm thinking something like that:
Would it make sense?

```fsharp
module ViewAttributes =
    let TextAttribKey = ...
    let TextColorAttribKey = ...

module ViewBuilders =
    let createLabel(...) = ...
    let buildLabel(...) = ...
    let updateLabel(...) = ...

module ViewHelpers =
    let getAttributeValue<'T> element key =
        match element.TryGetAttributeKeyed(key) with
        | None -> failwithf "Property '%s' does not exist on %s" key.Name element.TargetType.Name
        | Some attr -> attr.Value

type LabelViewer(element: ViewElement) =
    member val this.Text = ViewHelpers.getAttributeValue element ViewAttributes.TextAttribKey
    member val this.FontColor = ViewHelpers.getAttributeValue element ViewAttributes.FontAttribKey

type View() =
    static member Label(...) = ...
```

Usage:
```fsharp
let view model dispatch =
    View.ContentPage(
         content=View.StackLayout(
            children=[
               View.Label(text="Hello", textColor=Color.White)
            ]))

let testView() =
    let model = ...
    let view = MyApp.view model ignore
    let page = ContentPageViewer(view)
    let stackLayout = StackLayoutViewer(page.Content)
    let label = LabelViewer(stackLayout.Children.[0])
    label.Text |> should equal "Hello"
    label.Color |> should equal Color.White
```

**Comment by dsyme** [comment]
This looks reasonable, all things considered.

**Comment by TimLariviere** [comment]
Closed by #325
